home/views/admin.py

```diff
@@ -318,22 +318,17 @@ def verify_professor(self, verified_status: Professor.Status, slug: Optional[str
                     response["success_msg"] = "#info-modal-container"
                     return JsonResponse(response)
 
-                split_name = str(professor.name).strip().split(" ")
-                new_slug = split_name[-1].lower()
+                new_slug = create_autoslug(professor.name)
                 modal_msg = None
 
-                professors = Professor.verified.all()
-                if professors.filter(slug=new_slug).exists():
-                    new_slug = "_".join(reversed(split_name)).lower()
-                    if professors.filter(slug=new_slug).exists():
-                        modal_msg = mark_safe(f"The slug <b>{new_slug}</b> already belongs to a professor. Please enter a slug below.")
-
-                    if len(split_name) > 2:
-                        modal_msg = (
-                            f"The name '{professor.name}' is too long and "
+                if new_slug is None:
+                    modal_msg = (
+                        f"The name '{professor.name}' is too long and "
                             "can't be slugged automatcially. Please enter a slug below."
                         )
-
+                elif professors.filter(slug=new_slug).exists():
+                    modal_msg = mark_safe(f"The slug <b>{new_slug}</b> already belongs to a professor. Please enter a slug below.")
```

Comment:
There's no longer a concept of "too long to slug" with the new `create_slug` function.